dat/generated_tables.py

```diff
@@ -571,3 +571,86 @@ def create_iceberg_compat_v1(case: TestCaseInfo, spark: SparkSession):
     )
     delta_table.upgradeTableProtocol(3, 7)
     df.repartition(1).write.format("delta").mode("append").save(case.delta_root)
+
+
+# delta-io/delta golden tables migration
+def write_data(spark, data, mode, table_path):
+    schema = types.StructType([
+        types.StructField("col1", types.IntegerType(), True),
+        types.StructField("col2", types.StringType(), True)
+    ])
+    spark.createDataFrame(data, schema) \
+         .write \
+         .mode(mode) \
+         .format("delta") \
+         .save(table_path)
+
+def delete_data_with_condition(spark, table_path, condition):
+    table = DeltaTable.forPath(spark, table_path)
+    table.delete(condition)
+
+def repartition_table(spark, table_path, num_partitions):
+    df = spark.read.format("delta").load(table_path)
+    df.repartition(num_partitions) \
+      .write \
+      .option("dataChange", "false") \
+      .format("delta") \
+      .mode("overwrite") \
+      .save(table_path)
+
+@reference_table(name="snapshot-data0", description="golden tables snapshot-data0 test")
```

Comment:
i know these are the names from the golden tests in delta. they are _really_ not descriptive :) If you know a bit more what they are testing maybe we can rename them to add a little more color.